proto/p4/config/p4info.proto

```diff
@@ -55,6 +53,22 @@ message Preamble {
   repeated string annotations = 4;
 }
 
+// used to group all extern instances of the same type in one message
+message Extern {
+  // the extern_type_id is assigned during compilation. It is likely that this
+  // id will in fact come from a P4 annotation to the extern declaration and
+  // that each vendor will receive a prefix to avoid collisions.
```

Comment:
I don't expect that to be possible in P4-16 per se, but we are working on post-P4-16 extensions to the language that would allow that. I think we can be confident that a feature along those lines will be available.

It's true that we have only 32 bits for this field, which is too few to give high confidence that there will be no collisions. We could bump it up to 64 bits, I suppose, if we don't intend to transmit `extern_type_id` values very frequently in the API.